### PR TITLE
CI: mypy 0.990

### LINKED ISSUE #336: mypy 0.981 compatibility
mypy 0.981 released on 9/26/22.  If you do a manual re-run of a job, it picks that up, and we get some failures.  Could be mypy bugs, or settings, or errors we need to fix.

Might be able to remove the python 3.10.6 restriction, since I think 0.981 is supposed to fix the issue of using 3.10.7 or later.

### PR DESCRIPTION
**Pull request by twoertwein**
closes #336

xref #400

### CODE REVIEW

File: pandas-stubs/core/series.pyi [hunk 1]
```diff
@@ -456,7 +456,7 @@ class Series(IndexOpsMixin, NDFrame, Generic[S1]):
         dropna: _bool = ...,
     ) -> _SeriesGroupByNonScalar[S1]: ...
     @overload
-    def count(self, level: None = ...) -> int: ...
+    def count(self, level: None = ...) -> int: ...  # type: ignore[misc]
```
Comment (Dr-Irv) [inline pandas-stubs/core/series.pyi:459]: can you add a comment here as to why the `# type: ignore` is needed?

File: tests/test_io.py [hunk 2]
```diff
@@ -822,7 +822,7 @@ def test_sqlalchemy_selectable() -> None:
             class Base(metaclass=sqlalchemy.orm.decl_api.DeclarativeMeta):
                 __abstract__ = True
 
-            class Temp(Base):
+            class Temp(Base):  # type: ignore[misc]
```
Comment (Dr-Irv) [inline tests/test_io.py:825]: can you add a comment here as to why the `# type: ignore` is needed?

**Review by Dr-Irv — CHANGES_REQUESTED** [review]
Looks good.  Add some comments to explain some issues, and we should be good to go.

Can you also track which issues we can close because of this?  (if there are any such issues)

**Comment by bashtage** [comment]
Are most of these due to inheritance?

**Comment by twoertwein** [comment]
> Are most of these due to inheritance?

Yes, mypy seems now to be stricter about enforcing that a `property` is inherited by a `property`.

File: tests/test_io.py (hunk 2 above)
Comment (twoertwein) [reply tests/test_io.py:827]: I honestly don't understand why - I added the error reported by mypy

**Review by Dr-Irv — APPROVED** [review]
Thanks @twoertwein

**Pull request merged**